### PR TITLE
WW-351: Add tracking to Special:CreatePage

### PR DESCRIPTION
**Pull request by lukaszkonieczny**
@Wikia/west-wing

### CODE REVIEW

File: extensions/wikia/FlowTracking/scripts/createPageTracking.js [hunk 1]
```diff
@@ -45,7 +70,22 @@ define(
 			return namespaceId === 0;
 		}
 
+		function isTitleInMainNamespace(title) {
+			var namespace = '';
```
Comment (WikiaTech) [inline extensions/wikia/FlowTracking/scripts/createPageTracking.js]: MAJOR Remove this useless assignment to local variable "namespace" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 18 issues
* MAJOR 17 major
* INFO 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PageControls.js#L81](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L81): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L93](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L93): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L112](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L112): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L142](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L142): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L154](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L154): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L162](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L162): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L177](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L177): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L218](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L218): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L248](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L248): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L266](https://github.com/Wikia/app/blob/95ddf6a3535a6f6ea0334cda21c6a42a55204fa1/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L266): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* MAJOR 16 major
* INFO 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PageControls.js#L81](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L81): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L93](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L93): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L112](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L112): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L142](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L142): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L154](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L154): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L162](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L162): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L177](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L177): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L218](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L218): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L248](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L248): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [PageControls.js#L266](https://github.com/Wikia/app/blob/f357fca04577d0c98b7a6b371e8782cf06fb4212/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L266): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)

File: extensions/wikia/FlowTracking/scripts/createPageTracking.js [hunk 2]
```diff
@@ -45,7 +70,22 @@ define(
 			return namespaceId === 0;
 		}
 
+		function isTitleInMainNamespace(title) {
```
Comment (idradm) [inline extensions/wikia/FlowTracking/scripts/createPageTracking.js:73]: Isn't there any method that already does this?

File: extensions/wikia/FlowTracking/scripts/createPageTracking.js [hunk 3]
```diff
@@ -15,20 +15,45 @@ define(
...
 				qs.setVal('flow', window.wgFlowTrackingFlows.CREATE_PAGE_DIRECT_URL);
-			} else if (namespaceId === -1) {
-				// TODO: direct-url to Special:CreatePage (WW-351)
 			}
 
-			// set 'tracked' query param to prevent tracking the same event when page is reloaded
+			setTrackedQueryParam(qs);
+		}
+
+		function trackOnSpecialCreatePageLoad(editor) {
```
Comment (idradm) [inline extensions/wikia/FlowTracking/scripts/createPageTracking.js]: This method looks pretty similar to direct tracking method. Could we unify them somehow?

File: extensions/wikia/FlowTracking/scripts/createPageTracking.js (hunk 2 above)
Comment (lukaszkonieczny) [reply extensions/wikia/FlowTracking/scripts/createPageTracking.js:73]: I have not found anything

File: extensions/wikia/FlowTracking/scripts/createPageTracking.js (hunk 3 above)
Comment (lukaszkonieczny) [reply extensions/wikia/FlowTracking/scripts/createPageTracking.js]: One method is run for `namespace === 0` and the second for`namespace === -1```. Also conditions inside are different so I think it make sense to have them separate.

File: extensions/wikia/EditPageLayout/js/plugins/FlowTracking.js [hunk 4]
```diff
@@ -1,24 +1,40 @@
...
+				editor.on('changeTitle', this.proxy(this.onChangeTitle));
+			}
+		},
+
+		onChangeTitle: function (oldTitle, newTitle) {
+			var editor = this.editor;
+
+			require(['ext.wikia.flowTracking.createPageTracking'], function (flowTrackingCreatePage) {
+				// Check if title is provided first time and has valid namespace
+				if (oldTitle === '' && flowTrackingCreatePage.isTitleInMainNamespace(newTitle)) {
```
Comment (idradm) [inline extensions/wikia/EditPageLayout/js/plugins/FlowTracking.js]: Isn't it better to move namespace check into the tracking method? That way you wouldn't have to surface it.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
- MAJOR 16 major
- INFO 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [PageControls.js#L81](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L81): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
2. MAJOR [PageControls.js#L93](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L93): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
3. MAJOR [PageControls.js#L112](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L112): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
4. MAJOR [PageControls.js#L142](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L142): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
5. MAJOR [PageControls.js#L154](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L154): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
6. MAJOR [PageControls.js#L162](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L162): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
7. MAJOR [PageControls.js#L177](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L177): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
8. MAJOR [PageControls.js#L218](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L218): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
9. MAJOR [PageControls.js#L248](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L248): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
10. MAJOR [PageControls.js#L266](https://github.com/Wikia/app/blob/f11c7206c5ae11803e55a2290e6bc459a8e71610/extensions/wikia/EditPageLayout/js/plugins/PageControls.js#L266): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)

**Review by idradm — APPROVED** [review]

**Pull request merged**